src/config/configuration.validator.spec.ts

```diff
@@ -0,0 +1,33 @@
+import { faker } from '@faker-js/faker';
+import { validate } from './configuration.validator';
+
+describe('Configuration validator', () => {
+  const { NODE_ENV } = process.env;
+
+  afterAll(() => {
+    process.env.NODE_ENV = NODE_ENV;
+  });
+
+  it('should bypass this validation on tests', () => {
+    process.env.NODE_ENV = 'test';
+    const expected = JSON.parse(faker.datatype.json());
+    const validated = validate(expected);
+    expect(validated).toBe(expected);
+  });
+
+  it('should detect a malformed configuration in production environment', () => {
+    process.env.NODE_ENV = 'production';
```

Comment:
We are running tests in serial right now, to avoid cache collisions on the execution of the tests (this probably deserves a review also), so this test class takes care of restoring the original value of `process.env.NODE_ENV` (which should be `test`). But even if we switch back to running tests in parallel in the future, modifying the value of the keys in  `process.env` shouldn't be a cross-test problem since Jest provides a sandboxed environment for each test suite. 

(From [Jest docs](https://jestjs.io/docs/configuration#testenvironment-string))
```testEnvironment is sandboxed. Each test suite will trigger setup/teardown in their own TestEnvironment.```

Reply:
In fact, I think we could even remove the restoration for the same reason 🙂 I'll check that.

Reply:
I've removed the `process.env` restoration, Jest provides the encapsulated environment for each test. Let me know if you have any concern regarding this.